### PR TITLE
Update unit tests for configuration module

### LINKED ISSUE #5059: Rewrite unit tests for configuration module
## Description
Unit tests must be adapted to new RBAC and API implementation.

## Tasks

- [x] wazuh/core/cluster/tests/test_configuration.py

### PR DESCRIPTION
**Pull request by Selutario**
Hello team, this PR closes #5059 

## Description
It updates the unit tests to verify the correct working of the module `configuration.py`. Each function is tested individually and without the need to use a wazuh server.

Most tests were already done, but some things had to be changed to make it work with the new API. Coverage has also been increased to 100%.

### configuration.py
```
python3 -m pytest wazuh/tests/test_configuration.py
=================================================== test session starts ===================================================
platform linux -- Python 3.7.5, pytest-5.4.1, py-1.8.1, pluggy-0.13.1
rootdir: /home/selu/Git/wazuh/framework
plugins: tavern-1.0.0, cov-2.8.1
collected 32 items                                                                                                        

wazuh/tests/test_configuration.py ................................                                                  [100%]

==================================================== warnings summary =====================================================
```

```
----------- coverage: platform linux, python 3.7.5-final-0 -----------
Name                                             Stmts   Miss  Cover
--------------------------------------------------------------------
wazuh/InputValidator.py                             18     11    39%
wazuh/__init__.py                                   75     53    29%
wazuh/__main__.py                                    2      2     0%
wazuh/active_response.py                            20     20     0%
wazuh/agent.py                                     407    407     0%
wazuh/cdb_list.py                                   28     28     0%
wazuh/ciscat.py                                     26     26     0%
wazuh/cluster.py                                    43     43     0%
wazuh/common.py                                    101     13    87%
wazuh/configuration.py                             381      0   100%
wazuh/core/__init__.py                               0      0   100%
wazuh/core/active_response.py                       34     34     0%
wazuh/core/cdb_list.py                              52     38    27%
wazuh/core/cluster/__init__.py                       5      0   100%
wazuh/core/cluster/client.py                       148    148     0%
wazuh/core/cluster/cluster.py                      212    212     0%
wazuh/core/cluster/common.py                       333    333     0%
wazuh/core/cluster/control.py                       49     49     0%
wazuh/core/cluster/dapi/__init__.py                  0      0   100%
wazuh/core/cluster/dapi/dapi.py                    258    258     0%
wazuh/core/cluster/dapi/requests_list.py            20     20     0%
wazuh/core/cluster/local_client.py                  83     83     0%
wazuh/core/cluster/local_server.py                 131    131     0%
wazuh/core/cluster/master.py                       364    364     0%
wazuh/core/cluster/server.py                       152    152     0%
wazuh/core/cluster/tests/__init__.py                 0      0   100%
wazuh/core/cluster/tests/test_cluster.py            61     61     0%
wazuh/core/cluster/tests/test_common.py             28     28     0%
wazuh/core/cluster/tests/test_local_client.py       15     15     0%
wazuh/core/cluster/tests/test_worker.py             53     53     0%
wazuh/core/cluster/utils.py                        117     82    30%
wazuh/core/cluster/worker.py                       304    304     0%
wazuh/core/core_agent.py                           924    845     9%
wazuh/core/core_utils.py                            45     34    24%
wazuh/core/decoder.py                               44     44     0%
wazuh/core/manager.py                              140    140     0%
wazuh/core/rule.py                                 113     91    19%
wazuh/core/sca.py                                   28     28     0%
wazuh/core/security.py                               7      7     0%
wazuh/core/syscheck.py                              22     22     0%
wazuh/core/syscollector.py                          37     37     0%
wazuh/database.py                                   55     36    35%
wazuh/decoder.py                                    74     74     0%
wazuh/exception.py                                  76     31    59%
wazuh/manager.py                                   243    243     0%
wazuh/mitre.py                                      72     72     0%
wazuh/ossec_queue.py                                70     56    20%
wazuh/ossec_socket.py                               46     29    37%
wazuh/pyDaemonModule.py                             45     45     0%
wazuh/rbac/__init__.py                               0      0   100%
wazuh/rbac/auth_context.py                         166    166     0%
wazuh/rbac/decorators.py                           236    207    12%
wazuh/rbac/orm.py                                  679    679     0%
wazuh/rbac/preprocessor.py                          63     63     0%
wazuh/rbac/tests/__init__.py                         0      0   100%
wazuh/rbac/tests/test_auth_context.py               70     70     0%
wazuh/rbac/tests/test_decorators.py                 72     72     0%
wazuh/rbac/tests/test_default_configuration.py      62     62     0%
wazuh/rbac/tests/test_orm.py                       252    252     0%
wazuh/rbac/tests/test_preprocessor.py               27     27     0%
wazuh/rbac/tests/utils.py                           24     24     0%
wazuh/results.py                                   300    223    26%
wazuh/rootcheck.py                                 132    132     0%
wazuh/rule.py                                       84     84     0%
wazuh/sca.py                                        46     46     0%
wazuh/security.py                                  324    324     0%
wazuh/stats.py                                      96     96     0%
wazuh/syscheck.py                                   92     92     0%
wazuh/syscollector.py                               24     24     0%
wazuh/tests/test_configuration.py                  210      1    99%
wazuh/tests/util.py                                 33     23    30%
wazuh/utils.py                                     636    504    21%
wazuh/wdb.py                                       115     98    15%
wazuh/wlogging.py                                   66     50    24%
--------------------------------------------------------------------
TOTAL                                             9370   8121    13%
```

Kind regards,
José Luis.

### CODE REVIEW

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 5a28120f9d9041ca92417a114781b5646306e804
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2379/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2379/ossec.log)
**Coverage:** 

```json
Lines executed:3.93% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2379/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v4.0.0-B2379.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2379/Analysis_dynamic_valgrind-v4.0.0-B2379.log)

File: framework/wazuh/tests/test_configuration.py [hunk 1]
```diff
@@ -78,6 +80,12 @@ def test_read_option():
     with open(os.path.join(parent_directory, tmp_path, 'configuration/default/options1.conf')) as f:
         data = fromstring(f.read())
         assert configuration._read_option('labels', data)[0] == 'label'
+        assert configuration._read_option('test', data) == ('label', {'name': 'first', 'item': 'test'})
+
+    with open(os.path.join(parent_directory, tmp_path, 'configuration/default/options2.conf')) as f:
```
Comment (vicferpoy) [inline framework/wazuh/tests/test_configuration.py:85]:
```suggestion
    with open(os.path.join(parent_directory, tmp_path, 'configuration/default/synchronization.conf')) as f:
```
I suggest renaming this file to understand what it is simply reading its name.

File: framework/wazuh/tests/test_configuration.py [hunk 2]
```diff
@@ -292,9 +308,7 @@ def test_upload_group_file():
         with patch('wazuh.configuration.subprocess.check_output', return_value=True):
             with patch('wazuh.utils.chown', side_effect=None):
                 with patch('wazuh.utils.chmod', side_effect=None):
-                    assert configuration.upload_group_file('default', "<agent_config>new_config</agent_config>",
-                                                           'agent.conf') == \
-                           'Agent configuration was updated successfully'
+                    assert configuration.upload_group_file('default', "<agent_config>new_config</agent_config>", 'agent.conf') == 'Agent configuration was updated successfully'
```
Comment (vicferpoy) [inline framework/wazuh/tests/test_configuration.py:311]: Line is too long (PEP8)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** aa50a866a23e9c9c4ec90978d188406626ac708b
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2457/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2457/ossec.log)
**Coverage:** 

```json
Lines executed:3.93% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2457/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v4.0.0-B2457.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v4.0.0/B2457/Analysis_dynamic_valgrind-v4.0.0-B2457.log)

**Pull request merged**